### PR TITLE
Update sigs.k8s.io/bom to v0.6.0 and drop build for linux/s390x and windows/386

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

- Update sigs.k8s.io/bom to v0.6.0 and drop build for linux/s390x and windows/386

/assign @puerco @xmudrii @saschagrunert 
cc @kubernetes/release-managers 

from: https://github.com/kubernetes/release/pull/3419#issuecomment-1897800891
```
mmgrf the test is failing because `modernc.org/sqlite` which gets pulled in from the rpmdb module bom is using does not work in windows/386, not even when bumping it to the latest version :(
```

image

https://gitlab.com/cznic/sqlite/-/issues/169

So we are dropping the `windows/386` because of lack of support and also for `linux/s390x` which maybe we dont have any release managers using that platform

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
Update sigs.k8s.io/bom to v0.6.0 and drop build for linux/s390x and windows/386
```

### CODE REVIEW

**Comment by cpanato** [comment]
opened an issue for the deps review job to figure out what is going on: https://github.com/actions/dependency-review-action/issues/736

but this is a non-blocking job so we can merge that

/assign @puerco @xmudrii

File: hack/verify-build.sh [hunk 1]
```diff
@@ -24,9 +24,7 @@ PLATFORMS=(
     linux/arm
     linux/arm64
     linux/ppc64le
-    linux/s390x
     windows/amd64
-    windows/386
```
[the remark this answers is not in the document]
Comment (xmudrii) [reply hack/verify-build.sh:29]: Here in the PR is enough :)

[the remark this answers is not in the document]
Comment (cpanato) [reply hack/verify-build.sh:29]: done

[the remark this answers is not in the document]
Comment (xmudrii) [reply hack/verify-build.sh:29]: Thank you!

**Comment by saschagrunert** [comment]
@cpanato should we merge manually here?

**Comment by cpanato** [comment]
> @cpanato should we merge manually here?

i think so, lets do that

**Pull request merged**